### PR TITLE
feat(nwaku)_: receive messages via relay

### PR DESCRIPTION
**Pull request by richard-ramos**
Requires:
- https://github.com/waku-org/nwaku/pull/3205

Messages can be received via relay

### CODE REVIEW

**Review by gabrielmer — APPROVED** [review]
Amazingg, thanks so much!

Once it's merged, I'll update `waku-go-bindings` with the relevant parts

Maybe it would be useful to add a test to make sure message sending works

File: wakuv2/nwaku.go [hunk 1]
```diff
@@ -2448,6 +2474,31 @@ func newWakuNode(ctx context.Context, config *WakuConfig) (*WakuNode, error) {
...
 
+type jsonEvent struct {
+	EventType string `json:"eventType"`
+}
+
+func (n *WakuNode) OnEvent(eventStr string) {
+	jsonEvent := jsonEvent{}
+	err := json.Unmarshal([]byte(eventStr), &jsonEvent)
+	if err != nil {
+		return
```
Comment (Ivansete-status) [inline wakuv2/nwaku.go]: I wonder if we could log this error condition

File: wakuv2/nwaku.go [hunk 2]
```diff
@@ -2360,20 +2359,43 @@ func wakuNew(nodeKey *ecdsa.PrivateKey,
...
-	fmt.Println("Event received:", C.GoStringN(msg, C.int(len)))
+//export globalEventCallback
+func globalEventCallback(callerRet C.int, msg *C.char, len C.size_t, userData unsafe.Pointer) {
+	if callerRet == C.RET_OK {
+		eventStr := C.GoStringN(msg, C.int(len))
+		node, ok := nodeRegistry[userData]
+		if ok {
+			node.OnEvent(eventStr)
+		}
+	}
```
Comment (Ivansete-status) [inline wakuv2/nwaku.go]: Maybe we can handle the `else` condition to better control when a potential error occurs

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it! :100: 
I just added a couple of nitpick comments :)

**Pull request merged**